boto/provider.py

```diff
@@ -180,14 +183,24 @@ class Provider(object):
         }
     }
 
+    def __getstate__(self):
+        pickled_dict = copy.copy(self.__dict__)
+        del pickled_dict['_session']
+        return pickled_dict
```

Comment:
No, this is just to make pickle serialization works since the boto3 session object isnt serializable and isn't needed because it can be reconstructed.